experimental/parser/legalize_def.go

```diff
@@ -253,23 +253,33 @@ func legalizeMethod(p *parser, def ast.DeclDef) {
 		if sig.Inputs().Span().IsZero() {
 			def.MarkCorrupt()
 			p.Errorf("missing %v in %v", taxa.MethodIns, taxa.Method).Apply(
-				report.Snippet(def),
+				report.Snippetf(def.Name(), "expected type in %s after this", taxa.Parens),
 			)
 		} else {
 			legalizeMethodParams(p, sig.Inputs(), taxa.MethodIns)
 		}
 
 		if sig.Outputs().Span().IsZero() {
 			def.MarkCorrupt()
+			var after report.Spanner
+			switch {
+			case !sig.Returns().IsZero():
+				after = sig.Returns()
+			case !sig.Inputs().IsZero():
+				after = sig.Inputs()
+			default:
+				after = def.Name()
+			}
+
 			p.Errorf("missing %v in %v", taxa.MethodOuts, taxa.Method).Apply(
-				report.Snippet(def),
+				report.Snippetf(after, "expected type in %s after this", taxa.Parens),
```

Comment:
Looking at the error output below, I think this would be more clear if it said "return"
```suggestion
				report.Snippetf(after, "expected return type in %s after this", taxa.Parens),
```